src/main/java/com/avispl/symphony/dal/communicator/lg/lcd/LgLCDDevice.java

```diff
@@ -1350,59 +1784,103 @@ protected Object digestResponse(byte[] response, commandNames expectedResponse)
+						String tintValue = String.valueOf(Integer.parseInt(convertByteToValue(reply), 16));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.TINT, tintValue);
+						return tintValue;
 					case COLOR_TEMPERATURE:
+						reply = Arrays.copyOfRange(response, 7, 9);
+						String colorTemperature = String.valueOf(Integer.parseInt(convertByteToValue(reply), 16));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.COLOR_TEMPERATURE, colorTemperature);
+						return colorTemperature;
 					case BALANCE:
 						reply = Arrays.copyOfRange(response, 7, 9);
-						return String.valueOf(Integer.parseInt(convertByteToValue(reply), 16));
+						String balance = String.valueOf(Integer.parseInt(convertByteToValue(reply), 16));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.BALANCE, balance);
+						return balance;
 					case SOUND_MODE:
-						return EnumTypeHandler.getNameEnumByValue(SoundMode.class, convertByteToValue(reply));
+						String soundModeValue = EnumTypeHandler.getNameEnumByValue(SoundMode.class, convertByteToValue(reply));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.SOUND_MODE, soundModeValue);
+						return soundModeValue;
 					case NO_SIGNAL_POWER_OFF:
 						String noSignal = String.valueOf(Integer.parseInt(convertByteToValue(reply)));
 						String noSignalValue = LgLCDConstants.ON;
 						if (String.valueOf(LgLCDConstants.ZERO).equals(noSignal)) {
 							noSignalValue = LgLCDConstants.OFF;
 						}
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.NO_SIGNAL_POWER_OFF, noSignalValue);
 						return noSignalValue;
 					case NO_IR_POWER_OFF:
 						String noIRPower = String.valueOf(Integer.parseInt(convertByteToValue(reply)));
 						String noIRPowerValue = LgLCDConstants.ON;
 						if (String.valueOf(LgLCDConstants.ZERO).equals(noIRPower)) {
 							noIRPowerValue = LgLCDConstants.OFF;
 						}
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.NO_IR_POWER_OFF, noIRPowerValue);
 						return noIRPowerValue;
 					case LANGUAGE:
-						return EnumTypeHandler.getNameEnumByValue(Language.class, convertByteToValue(reply));
+						String languageValue = EnumTypeHandler.getNameEnumByValue(Language.class, convertByteToValue(reply));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.LANGUAGE, languageValue);
+						return languageValue;
 					case POWER_ON_STATUS:
-						return EnumTypeHandler.getNameEnumByValue(PowerStatus.class, convertByteToValue(reply));
+						String powerOnStatus = EnumTypeHandler.getNameEnumByValue(PowerStatus.class, convertByteToValue(reply));
+						updateCacheByValues(localCacheMapOfPropertyNameAndValue, LgLCDConstants.POWER_ON_STATUS, powerOnStatus);
+						return powerOnStatus;
 					default:
 						logger.debug("this command name is not supported" + expectedResponse);
```

Comment:
check for logger.isDebugEnable. no need to check for logger.isErrorEnabled(), the error is logged by default. please check for the similar errors in the communicator